graphql_compiler/cost_estimation/statistics.py

```diff
@@ -0,0 +1,211 @@
+            - None otherwise
+        """
+        return None
+
+
+class LocalStatistics(Statistics):
+    """Provides statistics using ones given at initialization."""
+    def __init__(self, class_count, edge_count_between_vertex_pair, domain_count, histogram):
+        """Initializes statistics with the given data.
+
+        Args:
+            class_count: dict, str -> int, mapping vertex/edge class name to class count.
+            edge_count_between_vertex_pair: dict, (str, str, str) -> int,
+                mapping tuple of (vertex out class name, vertex in class name, edge class name) to
+                count of edge type between vertex types.
+            domain_count: dict, (str, str) -> int, mapping vertex class and field on that
+                vertex class to domain size.
+            histogram: dict, (str, str) -> list[tuple(int, int, int)], mapping vertex class and
+                field on that vertex class to histogram as list[tuple(int, int, int)].
+        """
+        self._class_count = class_count
+        self._edge_count_between_vertex_pair = edge_count_between_vertex_pair
+        self._domain_count = domain_count
+        self._histogram = histogram
+
+    def get_class_count(self, class_name):
+        """Return how many objects have, or inherit, the given class name.
+
+        Args:
+            class_name: str, either a vertex class name or an edge class name defined in the
+                        GraphQL schema.
+
+        Returns:
+            - int, number of objects with, or inheriting the given class name
+
+        Raises:
+            AssertionError, if statistic for class_name does not exist.
+        """
+        if class_name not in self._class_count:
+            raise AssertionError(u'Class count statistic is required, but entry not found for: '
+                                 u'{}'.format(class_name))
+        return self._class_count[class_name]
+
+    def get_edge_count_between_vertex_pair(
+        self, vertex_out_class_name, vertex_in_class_name, edge_class_name
+    ):
+        """Return number of edges of given class connecting two vertex types.
+
+        For cost estimation, sometimes we need to predict traversal costs between two vertex types
+        using an edge type that connects the vertices' abstract types. If this statistic isn't
+        provided, it can be roughly estimated using multiple calls of get_class_count().
+
+        Args:
+            vertex_out_class_name: str, vertex class name.
+            vertex_in_class_name: str, vertex class name.
+            edge_class_name: str, edge class name.
+
+        Returns:
+            - int, the count of edges if the statistic exists
+            - None otherwise
+        """
+        statistic_id = (vertex_in_class_name, vertex_out_class_name, edge_class_name)
+        if statistic_id not in self._edge_count_between_vertex_pair:
+            return None
+        return self._edge_count_between_vertex_pair[statistic_id]
```

Comment:
nit: simplify to: `return self._edge_count_between_vertex_pair.get(statistic_id)`

Reply:
I'll push this in another PR shortly.